x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/api_key/30_update.yml

```diff
@@ -0,0 +1,355 @@
+---
```

Comment:
I'm going for happy path tests here since negative tests (e.g., non-owner or invalid updates) are covered quite thoroughly in `ApiKeyRestIT` and `ApiKeyIntegTests` already.

Reply:
The YAML tests look fine to me. That said, I just noticed two things in Java tests:
1. There is a bug in `ApiKeyIntegTests.testUpdateApiKeyAutoUpdatesUserRoles` in that the role definition most often do not change because the `putRoleWithClusterPrivileges` method [only honors the last cluster privilege](https://github.com/elastic/elasticsearch/blob/9016883e1c5fe1304b89394a1164531ea7cc50e9/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authc/ApiKeyIntegTests.java#L2048-L2050).
2. We don't seem to have tests for auto-update user information other than role definition changes, e.g. `full_name` or `email` changes. Role name change (as opposed to role definition changes) is not covered either.

Reply:
Ah, good catch(es)!

I already have partial tests for (2.) in the noop PR. I'll address the points you've identified in that PR. Hope that's all good.